src/Nuget.PackageIndex/NupkgLocalPackageLoader.cs

```diff
@@ -51,70 +78,84 @@ public IPackageMetadata GetPackageMetadataFromPath(string nupkgFilePath)
-                    //        (when assemblies under lib\fx\would have no types and use contract assemblies when they are referenced)
-                    //
-                    //      Logic for old nuget format:
-                    //       - if there is a lib\contract assembly - use only that assembly and ignore lib\fx subfolders
-                    //       - if there no lib\contract try to load types from all assemblies  in lib\fx folders
-                    //      Note: just in case we do check all dlls now to make sure there no lib\fx specific types in some
-                    //      assembies which are specific to that fx comparing to common types in a contract assembly.
-                    //      So dlls under contract would have all package's target frameworks, dlls under fx folder would have 
-                    //      only that particular target fx.
-                    //
-                    // New nuget format:
-                    //      - assemblies live under lib\fx folders as before
-                    //      - instdead of contract folder there is now Ref folder which contains common contracts. Unlice old contract
-                    //        folder, Ref folder might contain subfolder Any meaning that those contracts belong to any target fx; and
-                    //        corresponding Fx subfolder, which would contain types specific to particular FX.
-                    // 
-                    //      Logic for new format:
-                    //       - if there is a ref folder use assemblies under ref\any and then ref\fx; ignore lib assemblies
-                    //       - if there no ref folder just use assemblies from lib\fx subfolders
-                    //  
-                    // Note: just in case we process all assemblies lib, ref and contract. Assemblies with ref/contract that are in 
-                    // lib folder would not contain any types normally, so they shoudl be quick to process.
-
-                    var packageTargetFrameworks = package.GetSupportedFrameworks().Select(x => VersionUtility.GetShortFrameworkName(x));
+                    _nugetHelper.OpenPackage(fs);
+
+                    var packageFolder = Path.GetDirectoryName(nupkgFilePath) ?? string.Empty;
+                    var allAssemblies = _nugetHelper.GetPackageFiles()
+                                                    .Where(x => ".dll".Equals(Path.GetExtension(x.EffectivePath), StringComparison.OrdinalIgnoreCase)
+                                                                && !x.Path.ToLower().Contains(@"tools\")
+                                                                && !x.Path.ToLower().Contains(@"content\"));
+
+                    var hasContractAssemblies = false;
+                    var refAssemblies = allAssemblies.Where(x => x.Path.StartsWith("ref", StringComparison.OrdinalIgnoreCase));
```

Comment:
Actually I thought that paths inside of nupkg can be only standard folders like ref\xxx etc , but anyway I need to have a "ref\" there anyway